jax/core.py

```diff
@@ -328,11 +328,21 @@ def write(v, val):
       subfuns = [lu.wrap_init(partial(eval_jaxpr, call_jaxpr, ()))]
     else:
       subfuns = []
+    # TODO(apaszke): Make this initial -> final style conversion into a rule lookup
     if eqn.primitive.map_primitive:
       out_axes_thunk = HashableFunction(lambda: params['out_axes'],
                                         closure=params['out_axes'])
       bind_params = dict(params, out_axes_thunk=out_axes_thunk)
       del bind_params['out_axes']
+      if 'spmd_out_axes' in params:  # for xmap
```

Comment:
I already have a follow-up patch that makes this rule based instead of hardcoded in core. The way this works also doesn't work for pjit, which my next patch fixes.